File: src/openforms/js/components/admin/form_design/types/FormStep.js
import PropTypes from 'prop-types';

const FormStep = PropTypes.shape({
  configuration: PropTypes.object,
  formDefinition: PropTypes.string,
  index: PropTypes.number,
  name: PropTypes.string,
  internalName: PropTypes.string,
  slug: PropTypes.string,
  isApplicable: PropTypes.bool,
  loginRequired: PropTypes.bool,
  isReusable: PropTypes.bool,
  url: PropTypes.string,
  isNew: PropTypes.bool,
  validationErrors: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)),
  translations: PropTypes.objectOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      saveText: PropTypes.string.isRequired,
      previousText: PropTypes.string.isRequired,
      nextText: PropTypes.string.isRequired,
    })
  ),
});

export default FormStep;

File: src/openforms/forms/import_export/serializers/form_step.py
from openforms.forms.api.serializers import FormStepSerializer
from openforms.forms.models import FormVariable

from .base import BaseExportSerializer, BaseImportSerializer


class FormStepExportSerializer(FormStepSerializer, BaseExportSerializer):
    save_export_fields = (
        "uuid",
        "index",
        "slug",
        "configuration",
        "form_definition",
        "name",
        "internal_name",
        "url",
        "is_applicable",
        "login_required",
        "is_reusable",
        "literals",
        "previous_text",
        "save_text",
        "next_text",
        "translations",
    )


class FormStepImportSerializer(FormStepSerializer, BaseImportSerializer):
    def save(self, *args, **kwargs):
        super().save(*args, **kwargs)

        if (form := self.context.get("form")) is not None:
            # Once the form steps have been created, we create the component
            # FormVariables based on the form definition configurations.
            FormVariable.objects.create_for_form(form)
